packages/next/src/shared/lib/image-blur-svg.ts

```diff
@@ -16,20 +16,21 @@ export function getImageBlurSvg({
   blurDataURL: string
   objectFit?: string
 }): string {
-  const std = blurWidth && blurHeight ? '1' : '20'
+  const std = blurWidth && blurHeight ? 0.5 : 20
```

Comment:
Any reason why `1` was changed to `0.5`?

Did you take any screenshots comparing the two?

Reply:
Ok so for me there was quite a big visual difference between 1px and 20px blur placeholder images (1px blur images seemed to be too blurry). I started digging in to find out why and I've found this part right here

https://github.com/vercel/next.js/blob/946c9c5c46091604b8a9c73260bfe26ece70de23/packages/next/src/build/webpack/loaders/next-image-loader/blur.ts#L4
https://github.com/vercel/next.js/blob/946c9c5c46091604b8a9c73260bfe26ece70de23/packages/next/src/build/webpack/loaders/next-image-loader/blur.ts#L42-L54

https://github.com/vercel/next.js/blob/e17218fcba36ef148797dfefd5147e01cb29843b/packages/next/src/shared/lib/get-img-props.ts#L355-L356

So it looks like the "1px" blur is only used for blur placeholders generated by Next.js for staticly imported images (please correct me if I am wrong here) and they're at most 8px wide .

How I tried to figure out a better blur value for these small placeholders (8px wide/high maximum) is by doing simple calculations.

Taking two example images, one 390px wide (an iPhone 12 PRO screen width) and the second one 1512px wide (14" laptop screen width) I calculated the blur ratio bounds assuming the default blur is 20px.

20/1512 = 0.013 ----> 20/390 = 0.05 

Now translating that to 8px wide images:

0.013 * 8 = 0.1 ----> 0.05 * 8 = 0.4

Given that most traffic happens on mobile devices I've decided to go for the upper bound (images of smart phone screen width) and round it to 0.5 (because I know that SVG filters handle decimal halves values well).

## Alternative approach

Switching between different blur values depending on `blurWidth` and `blurHeight` being present seems a bit hacky to me. What we could do instead is to scale the `viewBox` attribute when `blurWidth` and `blurHeight` are set (since we know these values are at most 8px). Instead of doing `viewbox='0 0 8 5'` we could scale it by a 100 (to get closer to a normal image size) and do `viewbox='0 0 800 500'`. This way proportions are kept and blur stays consistent.

Reply:
@arturbien Sound good!

A few things:

> 390px wide (an iPhone 12 PRO screen width)

This is not accounting for the 3x DPR. You can confirm this by looking in dev tools emulation or by logging `window.devicePixelRatio` which prints `3`.

<img width="442" alt="image" src="https://github.com/vercel/next.js/assets/229881/f76d0a48-bfc7-4a7e-ad05-5bf8fd468d89">

So the 390px is really 1170px wide. 

Its quite common to have displays for 2x or 3x today, even on older devices.

> Instead of doing viewbox='0 0 8 5' we could scale it by 50 (to get closer to the normal image size ~ 400px ) and do viewbox='0 0 400 250'. This way proportions are kept and blur stays fixed at 20px.

That sounds like it could be a good approach, but where did you get the number `50`?

Reply:
> This is not accounting for the 3x DPR. You can confirm this by looking in dev tools emulation or by logging window.devicePixelRatio which prints 3.

AFAIK the viewBox doesn't 'care' about the DPI because it just specifies SVG dimensions in user space in px length unit (just like in CSS - width: 1px is actually 2 physical pixels wide for screens of DPI=2):

> The viewBox attribute defines the position and dimension, in user space, of an SVG viewport.

Also if DPI was problematic it would also have to break for the current viewBox/1px blur approach (As it remains constant across all possible DPIs)

> That sounds like it could be a good approach, but where did you get the number 50?

Well I just picked a 400px wide imaginary image for which 20px blur looks perfect. Then calculated the scale value 400px/8px (max blur width) = 50. The scale value of course can be tweaked but from my brief testing round it looked fine. 

Maybe I'll just implement that now, we'll test it and if you approve I'll update all tests so that we're ready to merge.

I think its worth a try because then we'll just have one simple filter that handles all cases no matter where the placeholder blur comes from 🤓

Reply:
Yeah lets try it out, as long as it works for all image sizes and not just the "400px wide imaginary image" 😄

Reply:
@styfle I've pushed the changes and fixed the blur value to 20px. Here's comparison for the `/fill-blur` demo page.
Other demos look good to, but take a second look and if you approve it I'll update all the test cases.

<img width="1018" alt="Screenshot 2023-07-24 at 09 00 38" src="https://github.com/vercel/next.js/assets/28541613/7187bf23-4629-45d7-a834-bed27a0f23b8">
<img width="1018" alt="Screenshot 2023-07-24 at 09 00 47" src="https://github.com/vercel/next.js/assets/28541613/a4ef8d08-f2bf-4869-bef8-f856ecffde0a">

Reply:
FYI I went with 40x multiplier as it looked better compared to 50x